doc/authoring_tests.md

```diff
@@ -77,6 +77,7 @@ Here are some issues that may occur when authoring tests that you should be awar
   2. check your parameter aliasing (particularly if it complains that a required parameter is missing that you know is there)
  If your command makes use of concurrency, consider using unit tests, LiveScenarioTest, or, if practical, forcing the test to operate on a single thread for recording and playback.
 * **Paths**: When including paths in your tests as parameter values, always wrap them in double quotes. While this isn't necessary when running from the command line (depending on your shell environment), it will likely cause issues with the test framework.
+* **Defaults**: Ensure you don't have any defaults configured with `az configure` prior to running tests. Defaults can interfere with the expected test flow.
```

Comment:
Nice! Totally forgot this but its good to include this gotcha.

Another gotcha is that when passing json data (which unfortunately some people might have done) one has to escape the dictionary with extra curly braces.

e.g:

```
self.cmd('relay hyco create -g testrg --namespace-name hcrealy --name hctest2 --user-metadata "[{\\"key\\":\\"endpoint\\",\\"value\\":\\"test:80\\"}]"')
```
Will lead to a key error.

Whereas the following succeeds:

```
self.cmd('relay hyco create -g testrg --namespace-name hcrealy --name hctest2 --user-metadata "[{{"key":"endpoint", "value":"vmsq1:80"}}]" ')
```

Not sure if its worth adding this, as it could inadvertently give command authors an excuse to attempt to use options that pass json data